File: src/main/java/cn/xunyard/idea/coding/log/LogMessage.java
package cn.xunyard.idea.coding.log;

import com.intellij.openapi.project.Project;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * @author <a herf="mailto:wuqi@terminus.io">xunyard</a>
 * @date 2019-12-26
 */
@Getter
@RequiredArgsConstructor
public class LogMessage {
    private final Project project;
    private final String identity;
    private final LogOperation operation;
    private final String format;
    private final Object[] arguments;
}

File: src/main/kotlin/cn/xunyard/idea/coding/log/window/LogMessage.kt
package cn.xunyard.idea.coding.log.window

import com.intellij.openapi.project.Project

/**
 * @author <a herf="mailto:xunyard@gmail.com">xunyard</a>
 * @date 2019-12-26
 */
data class LogMessage(
        val project: Project?,
        val identity: String ,
        val operation: LogOperation,
        val format: String?,
        val arguments: Array<out Any>
)